common/src/main/java/com/wynntils/core/config/ConfigManager.java

```diff
@@ -166,45 +172,33 @@ public static void saveDefaultConfig() {
         }
     }
 
-    private static List<ConfigHolder> collectConfigOptions(Feature feature) {
-        FeatureInfo featureInfo = feature.getClass().getAnnotation(FeatureInfo.class);
-        FeatureCategory category = featureInfo != null ? featureInfo.category() : FeatureCategory.UNCATEGORIZED;
-        feature.setCategory(category);
-        loadFeatureOverlayConfigOptions(category, feature);
-        return getConfigOptions(category, feature);
-    }
-
-    private static void loadFeatureOverlayConfigOptions(FeatureCategory category, Feature feature) {
-        // collect feature's overlays' config options
-        for (Overlay overlay : feature.getOverlays()) {
-            List<ConfigHolder> options = getConfigOptions(category, overlay);
-
-            registerConfigOptions(overlay, options);
+    private static Type findFieldType(Configurable parent, Field configField) {
+        Optional<Field> typeField = Arrays.stream(
+                        FieldUtils.getFieldsWithAnnotation(parent.getClass(), TypeOverride.class))
+                .filter(field ->
+                        field.getType() == Type.class && field.getName().equals(configField.getName() + "Type"))
+                .findFirst();
+
+        if (typeField.isPresent()) {
+            try {
+                return (Type) FieldUtils.readField(typeField.get(), parent, true);
+            } catch (IllegalAccessException e) {
+                WynntilsMod.error("Unable to get field " + typeField.get().getName(), e);
+            }
         }
+
+        return null;
     }
 
-    private static List<ConfigHolder> getConfigOptions(FeatureCategory category, Configurable parent) {
+    private static List<ConfigHolder> getConfigOptions(Configurable parent) {
         List<ConfigHolder> options = new ArrayList<>();
 
         for (Field configField : FieldUtils.getFieldsWithAnnotation(parent.getClass(), Config.class)) {
             Config metadata = configField.getAnnotation(Config.class);
 
-            Optional<Field> typeField = Arrays.stream(
-                            FieldUtils.getFieldsWithAnnotation(parent.getClass(), TypeOverride.class))
-                    .filter(field ->
-                            field.getType() == Type.class && field.getName().equals(configField.getName() + "Type"))
-                    .findFirst();
-
-            Type type = null;
-            if (typeField.isPresent()) {
-                try {
-                    type = (Type) FieldUtils.readField(typeField.get(), parent, true);
-                } catch (IllegalAccessException e) {
-                    WynntilsMod.error("Unable to get field " + typeField.get().getName(), e);
-                }
-            }
+            Type type = findFieldType(parent, configField);
```

Comment:
Can you rename this to indicate that this is a TypeOverride, and not the actual field type? Perhaps rename it to typeOverride, and findFieldTypeOverride